### ISSUE TITLE
Graph missing

### ISSUE THREAD

**Issue by deltaford** [nightscout/android-uploader#146]
No graph after upgrade from 0.1.10 to 0.1.12. Tried both mmol/l & mg/dl setting.

**Comment by rnpenguin** [comment]
what android OS are you running and what phone?

**Comment by deltaford** [comment]
We are using Android 4.4.4 (Cyganogenmod) on Moto G (G1 LTE) with Swedish locale.

**Comment by francescaneri** [comment]
We are using sansung nexus (Android 4.4.4 - Cyanogenmod) on Moto G (G1 LTE) with H3G ITA.

**Comment by trhodeos** [comment]
Can you guys take a screenshot and post it here?

**Comment by francescaneri** [comment]
![screenshot_2015-02-18-17-15-43](https://cloud.githubusercontent.com/assets/9915825/6251158/a067b012-b792-11e4-8556-e4f5bc8e09e7.png)

**Comment by trhodeos** [comment]
Thanks! I'll check this out later tonight.

**Comment by deltaford** [comment]
I'm seeing the exact same thing on my phone.

**Comment by scottleibrand** [comment]
This might be a timezone issue. I see what looks like one data point 4h
offset. Check that your Dexcom and Android phone both have the same time
and timezone.

**Comment by scottleibrand** [comment]
Well, Dexcom doesn't have a timezone per se, just a display time that needs
to match the uploader's.

**Comment by francescaneri** [comment]
controlled : are ok

**Comment by deltaford** [comment]
Dont know if it could be related but I had a similar issue (missing graph) with Chrome Uploader (https://github.com/nightscout/chrome-uploader/issues/16). That issue was related to (swedish) locale on the Dex.

"The dates from mongolab were coming across with a format of mm/dd/YYYY hh:mm:ss aa, BUT aa was 'fm' or 'em' instead of 'am'/'pm' and that was making it lose its mind.

My minute of googling didn't reveal what country uses those markers. Could you share?

In any case, I changed the date parsing code to try to handle it and got a graph after one last reset/download cycle."

**Comment by francescaneri** [comment]
https://francesca.azurewebsites.net

**Comment by francescaneri** [comment]
scruple changed the format of 12/24 hours both your device and on dex but nothing has changed

**Comment by oschumac** [comment]
Hi i hust have double checked on this also.
ON my phone Samsung galaxy express Android 4.1.2 and Samsung tab3 8 Android 4.4.2 .
Germay locale on both devices.
NightScout 0.1.12 no graph.  
After installing 0.1.10 the graph is back again.

CU Oliver

**Comment by francescaneri** [comment]
![screenshot](https://cloud.githubusercontent.com/assets/9915825/6576211/c95cfc90-c735-11e4-86d7-285d761cd06d.png)

I think the problem depends on what time of the European servers . It could at least be. Matteo

**Comment by francescaneri** [comment]
var coll = db.collection('test');
var datetime = new Date();
console.log(datetime);
coll.insert({ "t": "foo", "d": datetime}, function(err, result){
  if(err) throw err;
  coll.findOne({ "t" : "foo" }, function(err, doc) {
    console.log(doc.d);
  });
});

**Comment by francescaneri** [comment]
@trhodeos 

var coll = db.collection('test');
var datetime = new Date();
console.log(datetime);
coll.insert({ "t": "foo", "d": datetime}, function(err, result){
  if(err) throw err;
  coll.findOne({ "t" : "foo" }, function(err, doc) {
    console.log(doc.d);
  });
});

try checking. thank

**Comment by trhodeos** [comment]
What am I checking here?

**Comment by francescaneri** [comment]
@trhodeos ...for the problem of the graph. I asked for assistance in mongolab and a technician told me so I think it is a problem of European servers.
I state that no programmer.

**Comment by francescaneri** [comment]
I think it is a problem of European servers

**Comment by rnpenguin** [comment]
was that point just populated at the time shown from the computer?

**Comment by francescaneri** [comment]
@rnpenguin yes. if I move about 1 hour / 2 I see the graph . I did a test by changing the time and putting a + 2GMT already with the time zone of London you see the graph (android uploader)
time that there is exactly the time of mongolab London +.000 when it should be gmt + 1.00.

**Comment by alexpop** [comment]
Same problem over here.

Phone: Samsung Galaxy Note 4, Android 5.0.1
Phone timezone: GMT+01:00 British Summer Time
Dexcom: G4 Platinum, mmol/L version

Android app version 0.1.13: 
---
Dexcom and phone are using the correct time.

The app shows a blank graph like: https://github.com/nightscout/android-uploader/issues/146#issuecomment-74894142

The Nightscout web interface is showing the red 'in the future' label. It looks like the Dexcom data is uploaded with my current time + 1 hour.

To fix this, I've set the Dexcom device time an hour back. Fixes the uploader graph and the Nightscout web interface also.

Android app version 0.2.0-alpha10: 
---
With both phone and dexcom on the correct time, the graph is shifted one hour to the left. Nightscout web interface shows 'in the future'. With Dexcom time set one hour back everything is good.

THANK YOU!

**Comment by alexpop** [comment]
I updated my comment for 0.2.0-alpha20. Let me know if you want me to test a new release. Thanks

**Comment by alexpop** [comment]
I'm on GMT+1 and I saw this issue a few days ago. I must have installed version 0.1.12 or an earlier release.

Then, I uninstalled the app, installed 'Dreamsicle (Release 0.1.13)', put both dexcom and phone on the correct time. Looks good, no more graph gap and 'in the future' label. Thank you
